components/camel-jetty9/src/main/java/org/apache/camel/component/jetty9/JettyContentExchange9.java

```diff
@@ -214,14 +220,28 @@ public void onFailure(Request request, Throwable failure) {
             }
 
         };
-        BufferingResponseListener responseListener = new BufferingResponseListener() {
+
+        InputStreamResponseListener responseListener = new InputStreamResponseListener() {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+            @Override
+            public void onContent(Response response, ByteBuffer content, Callback callback) {
+                if (response.getStatus() != SC_OK) {
+                    LOG.warn("Response received {}: {}", response.getStatus(), response.getReason());
+                }
+                byte[] buffer = new byte[content.limit()];
```

Comment:
This will stream all the data into memory which is not what we want. Either we should use Camel's stream caching or some other kind of solution as people want to stream bigger data from jetty without having to have all the data in memory.

Camel has a stream caching that can overflow to disk for big data. Which we use in data formats etc and other places.

I wonder what others are doing with Jetty and this kind of listener.